src/odemis/driver/test/pwrmccdaq_test.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+'''
+Created on 21 Feb 2024
+
+@author: Stefan Sneep
+
+Copyright © 2024 Stefan Sneep, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+import copy
+import logging
+import time
+import unittest
+
+from odemis import model
+from odemis.driver import pwrmccdaq
+
+logging.getLogger().setLevel(logging.DEBUG)
+
+CONFIG_MCC = {
+                "name": "Interlock test", "role": "bsd-interlock",
+                "mcc_device": "fake",
+                "di_channels": {10: ["interlockTriggered", False]},
+              }
+
+CONFIG_LIGHT_DEVICE = {
+                "name": "Laser Hub", "role": "light",
+                "mcc_device": "fake",
+                "ao_channels": [0, 1],
+                "do_channels": [5, 6],
+                "spectra": [
+                    [592.e-9, 593.e-9, 594.e-9, 595.e-9, 596.e-9],
+                    [592.e-9, 593.e-9, 594.e-9, 595.e-9, 596.e-9]],
+                "pwr_curve": [
+                    {
+                        0: 0,
+                        5: 0.02,  # 20mW light emitter
+                    },
+                    {
+                        0: 0,
+                        5: 0.06,  # 60mW light emitter
+                    }],
+                "di_channels": {10: ["interlockTriggered", False]},
+}
+
+
+# django implementation of tag for use as a decorator in unittest (from django.test.utils)
```

Comment:
"django" ?! It's good that you indicate where this code comes from, but please first explain what this code does.

Reply:
Yes good point, at first it seemed quite handy to make a distinction between tests that use the setUp() and others that don't, and still keep them in the same class. Your suggestion below to use a separate private function to create the MCCDevice class fits this scenario better and I can leave out the "django" wrapper.